Api/CardRepositoryInterface.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Paytrail\PaymentService\Api;
+
+use Magento\Framework\Exception\LocalizedException;
+
+/**
+ * @api
+ */
+interface CardRepositoryInterface
```

Comment:
Consider renaming interface and implementation.

Expected methods and actions from a repository are:
- get($id) - load data from database by id.
- save($object) - Persist an object in the database.
- delete($object) - remove an object from database.
- deleteById($id) - load an object and then remove it from database.
- getList($searchCriteria) - allows searching for an object from database.

Your implementation does not match this pattern in Magento. Because of this I would recommend renaming your interface to "cardManagementInterface" or something similar to avoid confusion to third party developers who are going to use the module in projects.